oauth2-authorization-server/src/main/resources/org/springframework/security/oauth2/server/authorization/oauth2-authorization-schema.sql

```diff
@@ -0,0 +1,27 @@
+CREATE TABLE oauth2_authorization (
+    id varchar(100) NOT NULL,
+    registered_client_id varchar(100) NOT NULL,
+    principal_name varchar(100) NOT NULL,
+    authorization_grant_type varchar(100) NOT NULL,
+    attributes varchar(1000) DEFAULT NULL,
+    state blob DEFAULT NULL,
```

Comment:
For `state` and `attributes`: I'm wondering if it makes sense for `state` to be a `blob`, but all other attributes to be a `varchar(1000)`. Which flow would require a non-string state that is large enough to overflow 1000 chars? Would all other attributes fit comfortably in JSON format in 1000 characters?

If you have thought through that, awesome! Just thinking through this one myself, I don't have specific recommendations, but I'm wondering what your thoughts are.

Reply:
I think I left the state as blob because it is a base64 encoded string. Not sure if this could change in the future or not, but I see no issue with defining it as a `varchar(1000)`. For the attributes I think we should increase the size to `2000` or even more.  Let me know if this is ok and I will do the change in no time 👍

Reply:
Hi @ovidiupopa91, I think the `attributes` is fine, unless we know for sure we'll easily move past the 1000 limit. I suggest leaving it. Users can always increased the size suggested by this template. But I am wondering if we want to leave `state` as a blob, given it's a string (even base64 encoded). Perhaps switch it to `varchar(1000)` as you mentioned?